### PR TITLE
PW-404: Referral Handlers

### PR DESCRIPTION
**Pull request by alexnatalia**
# Task

[PW-404]: Referral system support via web app

## Changes

* Added Referral Handlers for History

[PW-404]: https://soramitsu.atlassian.net/browse/PW-404

### CODE REVIEW

**Review by RustemYuzlibaev — APPROVED** [review]

**Review by Nikita-Polyakov — APPROVED** [review]

**Review by pgolovkin — APPROVED** [review]

File: schema.graphql [hunk 1]
```diff
@@ -75,8 +75,32 @@ type ClaimableReward @entity {
...
+
+type ReferralReserve @entity {
+	id: ID!
+	blockHeight: BigInt!
+	address: String! @index
+	timestamp: Int!
+	amount: String!
+}
+
+type ReferralUreserve @entity {
```
Comment (bgodlin) [inline schema.graphql]: A small typo here

File: src/mappings/referrerRewardHandler.ts [hunk 2]
```diff
@@ -5,9 +5,10 @@ import { formatU128ToBalance } from "./utils";
 export async function referrerRewardHandler(
 	event: SubstrateEvent
 ): Promise<void> {
+
 	const {
 		event: {
-			data: [referree, referrer, amount],
+			data: [referral, referrer, amount],
```
Comment (bgodlin) [inline src/mappings/referrerRewardHandler.ts]: Seems like we have a definition error in https://soramitsu.atlassian.net/browse/PW-404

File: schema.graphql [hunk 3]
```diff
@@ -75,8 +75,32 @@ type ClaimableReward @entity {
...
 	blockHeight: BigInt!
-	referrer: String! @index
-	referree: String!
+	referral: String! @index
+	referrer: String!
+	timestamp: Int!
+	amount: String!
+}
+
+type SetReferral @entity {
```
Comment (bgodlin) [inline schema.graphql]: All new added entities are not used in the mapping files - then, I guess, there is no need adding the here

**Review by bgodlin — CHANGES_REQUESTED** [review]
All new added `.ts` files will not be handling as there are not appended to the `src/mapping/index.ts`. Still, should we aggregate all functions in new files to the single file, called, for instance, `"referralSystemHandlers.ts"`?

Additionally, all new extrinsics will be saved as the HistoryElement entity, which will be storing all other handled extrinsics (transafers, swaps and other displayed at `#account` page) in the same table at the db level - thus, there may be performance issues. Thus, I suppose, to increase the performance, it would be reasonable to assign a separate entity for every entity that will have a separate view in final UI. For instance, if we're having a view that will list of reserve and unreserve activities, the most optimized way will be to group the corresponding extrinsics in the separate entity.

**Review by bgodlin — APPROVED** [review]

**Pull request merged**